src/object.rs

```diff
@@ -254,11 +256,17 @@ impl Object {
     /// # Ok(())
     /// # }
     /// ```
-    pub async fn range_read(&self, range: impl RangeBounds<u64>) -> Result<Vec<u8>> {
-        let op = OpRead::new(self.path(), range)?;
+    pub async fn range_read(&self, range: impl RangeBounds<u64> + Clone) -> Result<Vec<u8>> {
+        let op = OpRead::new(self.path(), range.clone())?;
         let s = self.acc.read(&op).await?;
 
-        let mut bs = Cursor::new(Vec::new());
+        let br = BytesRange::from(range);
+        let buffer = if let Some(range_size) = br.size() {
+            Vec::with_capacity(std::cmp::min(range_size as usize, 128 * 1024 * 1024))
```

Comment:
OpenDAL's user does know what they do, pass size directly here is good to me.

And this can't help the user: they will OOM in `copy` anyway.